base/base.jl

```diff
@@ -82,7 +82,7 @@ finalize(o::ANY) = ccall(:jl_finalize, Void, (Any,), o)
 gc(full::Bool=true) = ccall(:jl_gc_collect, Void, (Cint,), full)
 gc_enable(on::Bool) = ccall(:jl_gc_enable, Cint, (Cint,), on)!=0
 
-bytestring(str::ByteString) = str
+bytestring(str::String) = str
```

Comment:
I started down that road, but that's a very big sweater once you start pulling the strings, and this change plus that one were just too much when combined.